synapse/lib/base.py

```diff
@@ -156,17 +156,6 @@ async def enter_context(self, item):
         Returns:
             The result of item’s own __aenter__ or __enter__() method.
         '''
-        async def fini():
```

Comment:
We have apparently been dup-calling exit handlers for all Base.enter_context managed objects?!?!?